### PR TITLE
Move try catch to test base

### PR DESCRIPTION
**Pull request by vasilytric**
Moving `try-catch` common blocks from test to tests base
I did not move this blocks from `opencl` tests or non common blocks

This PR contain many changes due to code was reformatted after removing `try-catch` blocks. 
Ronan Keryell offered to do it in [this](https://github.com/KhronosGroup/SYCL-CTS/pull/140#discussion_r681945924) comment

### CODE REVIEW

**Review by ProGTX — APPROVED** [review]

File: util/test_base.cpp
Comment (keryell) [inline util/test_base.cpp:26]: As I suggested in another PR, what is important is that an exception was thrown.
Of course to know it we need to catch the exception, but this is an implementation detail.
```suggestion
    auto errorMsg = "a SYCL exception was thrown: " + std::string(e.what());
    FAIL(log, errorMsg);
  } catch (const std::exception &e) {
    auto errorMsg = "an exception was thrown: " + std::string(e.what());
```

File: util/test_base.h
Comment (keryell) [inline util/test_base.h:48]:
```suggestion
  /** member function that will be overridden in test file
```

File: util/test_base.h
Comment (keryell) [inline util/test_base.h:53]:
```suggestion
  /** run overridden member function with try-catch block
```

**Review by keryell — CHANGES_REQUESTED** [review]
This is quite an achievement! \o/
You have removed +8,330 −9,655 = 1325 lines of code.
Go on like that! :-)
Just a few remarks at the end.

**Comment by psalz** [comment]
While I agree that tests absolutely shouldn't have to be manually wrapped in `try-catch` blocks, I'm not sure whether it is worth retroactively changing this for all of the existing tests, obliterating `git blame` for most of the CTS. Couldn't we just wrap ` test_base::run` in a `try-catch` internally, so that new tests don't have to do it manually, while existing tests simply continue to catch the exceptions themselves?

**Comment by fraggamuffin** [comment]
Sept 9:
* ok to merge when it is ready

**Comment by ProGTX** [comment]
> 
> 
> While I agree that tests absolutely shouldn't have to be manually wrapped in `try-catch` blocks, I'm not sure whether it is worth retroactively changing this for all of the existing tests, obliterating `git blame` for most of the CTS. Couldn't we just wrap ` test_base::run` in a `try-catch` internally, so that new tests don't have to do it manually, while existing tests simply continue to catch the exceptions themselves?

As mentioned, if we remove the `try-catch` blocks while still keeping the original scope (curly braces) of the `try` block, none of that gets changed, so `git blame` still shows the proper history.

**Comment by psalz** [comment]
> As mentioned, if we remove the `try-catch` blocks while still keeping the original scope (curly braces) of the `try` block, none of that gets changed, so `git blame` still shows the proper history.

Ah, I misunderstood you then. That's actually a neat idea!

**Comment by keryell** [comment]
> As mentioned, if we remove the `try-catch` blocks while still keeping the original scope (curly braces) of the `try` block, none of that gets changed, so `git blame` still shows the proper history.

Otherwise we can just postpone the `clang-format` phase up to the future, so with a local indent of 4 instead of 2.

**Comment by psalz** [comment]
> > As mentioned, if we remove the `try-catch` blocks while still keeping the original scope (curly braces) of the `try` block, none of that gets changed, so `git blame` still shows the proper history.
> 
> Otherwise we can just postpone the `clang-format` phase up to the future, so with a local indent of 4 instead of 2.

This is what I originally though @ProGTX was proposing. It would suddenly introduce a lot more unformatted files into the codebase, which is something I think we should always try to avoid.

File: util/test_base.cpp
Comment (keryell) [reply util/test_base.cpp:26]: This was done in https://github.com/KhronosGroup/SYCL-CTS/pull/143 which is now merged in. So there is just some homogenizing work here.

**Review by keryell — COMMENTED** [review]
When it was discussed during a recent committee meeting, the consensus was to merge this once the merge conflicts are solved.
Thanks for the cleaning!

**Comment by bader** [comment]
@vasilytric, could you resolve merge conflicts, please?

**Comment by bader** [comment]
@psalz, FYI. CI in works.

**Comment by vasilytric** [comment]
> @vasilytric, could you resolve merge conflicts, please?

@bader I've done it in this merge-commit: 8c87087f537260069b018b82c34989b07350e351 
but got unexpected error, looks like this error is not related to code from this PR, but related to different library (`hipsycl`)

**Comment by bader** [comment]
> > @vasilytric, could you resolve merge conflicts, please?
> 
> @bader I've done it in this merge-commit: [8c87087](https://github.com/KhronosGroup/SYCL-CTS/commit/8c87087f537260069b018b82c34989b07350e351)
> but got unexpected error, looks like this error is not related to code from this PR, but related to different library (`hipsycl`)

Great, it means the notification works as expected. :)
Please, ignore CTS build failures with hipsycl implementation. @psalz is planning to address them in a separate pull request.

**Comment by vasilytric** [comment]
> Great, it means the notification system works as expected. :)

Thanks for the information :)

File: util/test_base.h
Comment (vasilytric) [reply util/test_base.h]: I've applied this comment in this commit: 4a4aa32e58ee565190fd5c4ff444133660b1ea7b

File: util/test_base.h
Comment (vasilytric) [reply util/test_base.h]: I've applied this comment in this two commits: 4a4aa32e58ee565190fd5c4ff444133660b1ea7b and 54f529b51c7864445834a0e62af510481bf755cf

**Review by keryell — APPROVED** [review]
Great!

File: tests/invoke/invoke_kernel_params.cpp
Comment (psalz) [inline tests/invoke/invoke_kernel_params.cpp:75]: Merge conflict marker

File: tests/invoke/invoke_kernel_params.cpp
Comment (psalz) [inline tests/invoke/invoke_kernel_params.cpp:154]: Merge conflict marker

**Review by psalz — CHANGES_REQUESTED** [review]
Case in point that this patch may be a bit too large to review: Currently it contains two merge conflict markers.

File: tests/invoke/invoke_kernel_params.cpp
Comment (vasilytric) [reply tests/invoke/invoke_kernel_params.cpp]: Thanks for the pay attention, I've removed this in this commit: f6cd54a48b76cbfb3cd2ad739c05ed42d43dcd3c

File: tests/invoke/invoke_kernel_params.cpp
Comment (vasilytric) [reply tests/invoke/invoke_kernel_params.cpp]: Thanks for the pay attention, I've removed this in this commit: f6cd54a48b76cbfb3cd2ad739c05ed42d43dcd3c

**Comment by bader** [comment]
@vasilytric, please, resolve merge conflicts with recent commits to the target branch.

**Comment by psalz** [comment]
I'd like to again entertain @ProGTX's idea of keeping the current scope to avoid excessive reformatting. This line gets us there 99% of the way: `find tests -type f | xargs gawk -i inplace '/try {/ { print "{"; next } /} catch \(/ { skip = 1; print "}"; next }; skip == 1 && /}$/ { skip = 0; next } skip == 1 { next } 1'; git diff -U0 --no-color | clang-format-diff-10 -p1 -i`.

I'd be willing to file a PR containing this change plus some cherry-picked patches by @vasilytric such as 531159c.

**Comment by vasilytric** [comment]
> @vasilytric, please, resolve merge conflicts with recent commits to the target branch.

@bader merge conflicts was resolved in this commit: 377e4cb84efb2b8bf573ff82d19a90e90d92c6eb

**Comment by bader** [comment]
Superseded by #190.

**Pull request closed without merging**
